### ISSUE TITLE
Основные настройки страницы

### ISSUE THREAD

**Issue by DmitryBorisenko33** [tretyakovsa/Sonoff_WiFi_switch#159]
Здравствуйте. У меня возникли затруднения с основными настройками страницы. 

```
{
 "configs": [
  "/config.live.json",
  "/config.option.json",
  "/config.setup.json"
 ],
 "class":"col-sm-offset-1 col-sm-10 col-md-offset-2 col-md-8 col-lg-offset-3 col-lg-6",
 "content": [
  {
   "type": "h5",
   "title": "{{SSDP}}",
   "class":"alert-warning"
  },
{
   "type": "h4",
   "title": "Device ID: {{chipID}}"
  },
{
   "type": "h4",
   "title": "IP address: {{ip}}"
  },
{
   "type": "hr"
 },
{
   "type": "link",
   "title": "Сценарии",
   "action": "/page.htm?scenario",
   "class": "btn btn-block btn-primary"
  },
  {
   "type": "link",
   "title": "Главная",
   "action": "/page.htm?index",
   "class": "btn btn-block btn-danger btn-sm"
  }
 ]
}
```

Вот это код моей страницы. Она состоит из блока который размещен по центру экрана. Я хотел бы научиться изменять размер этого блока по горизонтали. Я пытался редактировать класс но не получилось. 
![Screenshot_1](https://user-images.githubusercontent.com/49808844/56793436-9f37b500-683e-11e9-8960-dda7a0a66a05.png)

**Comment by renat2985** [comment]
То есть вы хотите весь главный блок сделать других размеров?
Для этого вам нужно в самом верху играться с этой строчкой:
`"class":"col-sm-offset-1 col-sm-10 col-md-offset-2 col-md-8 col-lg-offset-3 col-lg-6",`
Можете попробовать заменить на это, блок станет значительно больше:
`"class":"col-sm-offset-1 col-sm-10",`
Так же можно этот блок сдвинуть в левую или правую часть на нужное расстояние. Почитайте о Bootstrap Grid.

Но есть 1 момент. К сожалению все что находится вне content":[*] не меняется динамически, то есть нажав на VIEW вы не увидите изменений. Нужно нажать SAVE и обновить страницу.

**Comment by DmitryBorisenko33** [comment]
Описания Bootstrap Grid приведены в этой ссылке: https://getbootstrap.com/docs/4.1/examples/grid/
Существует несколько типов отображения страниц: 

sm для устройств ≥576px, 
md для устройств ≥768px, 
lg для устройств ≥992px, 
xl для устройств ≥1200px. 

Это значит что когда мы пишем строку:

"class":"
col-sm-offset-1 col-sm-10 
col-md-offset-2 col-md-8 
col-lg-offset-3 col-lg-6",

то мы задаем пропорции экрана для каждого типа устройств в отдельности.

Можно увидеть что отступ offset растет с увеличением количества пикселей. Что логично потому что он не нужен большим на устройствах с маленьким экраном. 

Еще очень важный нюанс в том что если я напишу в класс только sm
"col-sm-offset-1 col-sm-10" 
то устройство поделит экран по стандарту sm для всех устройств принудительно даже для тех у кого будет ширина экрана больше ≥576 выглядеть это будет так:

![1](https://user-images.githubusercontent.com/49808844/56832629-1b61e500-689f-11e9-8a99-b71892584684.png)
 
Если я напишу класс так(sm md): 
"col-sm-offset-1 col-sm-10 col-md-offset-2 col-md-8"
то устройство поделит экран по стандарту md для устройств у которых поперечное разрешение больше  ≥768px:

![2](https://user-images.githubusercontent.com/49808844/56832994-e5713080-689f-11e9-9472-387cc2d00729.png)

Если я напишу класс так(sm md lg): 
"col-sm-offset-1 col-sm-10  col-md-offset-2 col-md-8 col-lg-offset-3 col-lg-6"
то устройство поделит экран по стандарту lg для устройств у которых поперечное разрешение больше  ≥992px:

![3](https://user-images.githubusercontent.com/49808844/56833353-d76fdf80-68a0-11e9-9e36-fdbc5bdabbcb.png)

то есть вывод такой:

576px > sm > 768px
768px > md > 992px
992px > lg > 1200px

Из описания я понял что весь экран делится на 12 частей. И если ты пишешь col-sm-12 то значит блок будет во всю ширину. Если напишешь к примеру col-sm-6 то это половина из возможной ширины. 
Параметр col-sm-offset-1 будет значить что я оставлю на офсет 1/12 полной ширины страницы.

**Comment by renat2985** [comment]
Да, все правильно. Спасибо за подробное описание, наверняка кому-то пригодится.
Есть много вспомогательных сайтов которые помогаю сделать нужные размеры на разных дивайсах. Мне нравится вот этот http://shoelace.io/ С право выбираешь устройство, по серединке размеры блока на этом устройстве и в конце с лева копируешь полученный class

**Comment by DmitryBorisenko33** [comment]
Я воспользовался предложенным сайтом. Очень крутая вещь.
Вышло вот так для устройств Desktop:
![1](https://user-images.githubusercontent.com/49808844/56846204-fb620e00-68fe-11e9-9939-739a904d2df6.png)
Тут же можно увидеть как это будет выглядеть на телефоне:
![2](https://user-images.githubusercontent.com/49808844/56846230-6f041b00-68ff-11e9-91c6-a66a094f6132.png)

**Comment by DmitryBorisenko33** [comment]
Теперь хотелось бы спросить о размещении  header title и footer на странице.

Для проверки я добавил вот такой код на страницу:

```
 "title":[
{
   "type": "h2",
   "title": "MY TITLE"
  }
],
```
Получил вот такой результат:

![3](https://user-images.githubusercontent.com/49808844/56846358-03bb4880-6901-11e9-8f56-3b647a3d2688.png)

смотрится очень круто.

Хотелось спросить а как создать несколько блоков рядом?

**Comment by renat2985** [comment]
Посмотрите вот это мое видео https://youtu.be/HcyibmxBZio

**Comment by alecseys** [comment]
Доброго времени суток!
У меня возникла проблема при попытки использовать title cсовместно с socket. После каждого нового поступления данных через socket генерится новая строка в блоке title. В чем моя ошибка? 
![image](https://user-images.githubusercontent.com/56131628/66152867-0e51da80-e623-11e9-97b7-4e06f55a5ac3.png)

**Comment by alecseys** [comment]
В связи с тем, что заголовок в блоке title предполагался статический, решил вопрос, путем корректировки файла index.html.gz.

**Comment by reo1978** [comment]
Здравствуйте подскажите как изменить фон(цвет) контейнера в котором располагается контент  на странице управления например?

**Comment by renat2985** [comment]
@reo1978 
![DeepinScreenshot_выберите-область_20200125095932](https://user-images.githubusercontent.com/1230815/73118408-64dd7e80-3f5c-11ea-9f8f-e630df3284e8.png)
